MapboxNavigation/SystemSpeechSynthesizer.swift

```diff
@@ -48,20 +48,28 @@ open class SystemSpeechSynthesizer: NSObject, SpeechSynthesizing {
     
     // MARK: - Public methods
     
-    open func prepareIncomingSpokenInstructions(_ instructions: [SpokenInstruction]) {
+    open func prepareIncomingSpokenInstructions(_ instructions: [SpokenInstruction], locale: Locale?) {
         // Do nothing
     }
     
-    open func speak(_ instruction: SpokenInstruction, during legProgress: RouteLegProgress) {
+    open func speak(_ instruction: SpokenInstruction, during legProgress: RouteLegProgress, locale: Locale? = nil) {
         guard !muted else {
             delegate?.speechSynthesizer(self,
                                         didSpeak: instruction,
                                         with: nil)
             return
         }
         
+        guard let locale = locale ?? self.locale else {
+            self.delegate?.speechSynthesizer(self,
+                                             encounteredError: SpeechError.undefinedSpeechLocale(instruction: instruction))
+            return
+        }
+        
         var utterance: AVSpeechUtterance?
-        if Locale.preferredLocalLanguageCountryCode == "en-US" {
+        let localeCode = [locale.languageCode, locale.regionCode].compactMap{$0}.joined(separator: "-")
+        
+        if localeCode == "en-US" {
```

Comment:
It would be cleaner to check if `locale.languageCode == "en" && locale.regionCode == "US"`.